src/data/sym/basic.lean

```diff
@@ -324,36 +326,45 @@ protected def cast {n m : ℕ} (h : n = m) : sym α n ≃ sym α m :=
 
 @[simp] lemma cast_rfl : sym.cast rfl s = s := subtype.ext rfl
 
-@[simp] lemma cast_cast {n n' n'' : ℕ} (s : sym α n) (h : n = n') (h' : n' = n'') :
+@[simp] lemma cast_cast {n'' : ℕ} (h : n = n') (h' : n' = n'') :
   sym.cast h' (sym.cast h s) = sym.cast (h.trans h') s := rfl
 
-@[simp] lemma coe_cast {n m : ℕ} (h : n = m) (s : sym α n) :
-  (sym.cast h s : multiset α) = s := rfl
+@[simp] lemma coe_cast  (h : n = m) : (sym.cast h s : multiset α) = s := rfl
+
+@[simp] lemma mem_cast (h : n = m) : a ∈ sym.cast h s ↔ a ∈ s := iff.rfl
 
 /-- Append a pair of `sym` terms. -/
-def append {n n' : ℕ} (s : sym α n) (s' : sym α n') : sym α (n + n') :=
+def append (s : sym α n) (s' : sym α n') : sym α (n + n') :=
 ⟨s.1 + s'.1, by simp_rw [← s.2, ← s'.2, map_add]⟩
 
-@[simp] lemma append_inj_right {n n' : ℕ} (s : sym α n) (t t' : sym α n') :
+@[simp] lemma append_inj_right (t t' : sym α n') :
```

Comment:
Maybe keep some of these arguments here and elsewhere to manage which are explicit and which are implicit?
```suggestion
@[simp] lemma append_inj_right (s : sym α n) {t t' : sym α n'} :
```
I'm following the convention for `add_right_inj` here.

Reply:
I don't really have a good idea what should be explicit/implicit. What I did for now is make anything that was in variables implicit, except for the more difficult lemmas, where I left them as explicit. Do you have any other specific ones in mind that should be reworked for consistency/other reasons?

Reply:
I'm assuming that the lemmas that prove these have been thought about, which is why I suggested following `add_right_inj`'s convention here. We should do the same for all the other lemmas unless we have reason not to. I'll flag the ones I notice.